modular_skyrat/modules/ashwalkers/code/buildings/ash_farming.dm

```diff
@@ -1,141 +1,209 @@
-/turf/open/misc/asteroid/basalt/lava_land_surface/attackby(obj/item/attacking_object, mob/user, params)
-	if(istype(attacking_object, /obj/item/grown))
-		var/obj/item/grown/attacking_grown = attacking_object
-		if(attacking_grown.seed)
-			try_seeding(attacking_grown.seed, attacking_grown, user)
-			return
-	else if(istype(attacking_object, /obj/item/food/grown))
-		var/obj/item/food/grown/attacking_food = attacking_object
-		if(attacking_food.seed)
-			try_seeding(attacking_food.seed, attacking_food, user)
-			return
+/datum/component/simple_farm
+	///whether you can actually farm it at the moment
+	var/allow_plant = FALSE
+	///whether we limit the amount of plants you can have per turf
+	var/one_per_turf = TRUE
+	///the reference to the movable parent the component is attached to
+	var/atom/atom_parent
```

Comment:
doesn't seem needed

Reply:
not needed, but allows people to create variants that can plant multiple plants per turf.